integrationTests/testInitializer.go

```diff
@@ -816,6 +817,35 @@ func CreateNodes(
 	return nodes
 }
 
+// CreateNodesWithMemP2P creates multiple nodes in different shards
+func CreateNodesWithMemP2P(
+	numOfShards int,
+	nodesPerShard int,
+	numMetaChainNodes int,
+	network *memp2p.Network,
+) []*TestProcessorNode {
+	//first node generated will have is pk belonging to firstSkShardId
+	nodes := make([]*TestProcessorNode, numOfShards*nodesPerShard+numMetaChainNodes)
+
+	idx := 0
+	for shardId := uint32(0); shardId < uint32(numOfShards); shardId++ {
+		for j := 0; j < nodesPerShard; j++ {
+			n := NewTestProcessorNodeWithMemP2P(uint32(numOfShards), shardId, shardId, network)
+
+			nodes[idx] = n
+			idx++
+		}
+	}
+
+	for i := 0; i < numMetaChainNodes; i++ {
```

Comment:
This would be clearer:
```
numValidators := len(nodes)
for i := i < numMetaChainNodes; i++ {
    metaNode := ...
    nodes[i + numValidators] = metaNode
}
```

Reply:
Something is not right with the for loop condition